### PR TITLE
feat: Issue Credential - state implementation for proposal sent/received

### PR DESCRIPTION
**Pull request by soluchok**
* `HandleOutbound` function can send a proposal.
* `HandleInbound` function can receive a proposal.

Part of #1173 
Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: test/bdd/go.mod [hunk 1]
```diff
@@ -8,6 +8,7 @@ go 1.14
 
 require (
 	github.com/Microsoft/hcsshim v0.8.7 // indirect
+	github.com/btcsuite/btcutil v1.0.1
```
Comment (Baha-sk) [inline test/bdd/go.mod]: why is this line added in this PR?

Comment (soluchok) [reply test/bdd/go.mod]: Because someone missed to add it. I’ve just added it into my PR.

File: pkg/didcomm/protocol/issuecredential/states.go [hunk 2]
```diff
@@ -48,6 +54,9 @@ type state interface {
 	ExecuteOutbound(msg *metaData) (state, stateAction, error)
 }
 
+// represents zero state's action
+func zeroAction(service.Messenger) error { return nil }
```
[the remark this answers is not in the document]
Comment (rolsonquadras) [reply pkg/didcomm/protocol/issuecredential/states.go]: looks like this is used for no action.

[the remark this answers is not in the document]
Comment (soluchok) [reply pkg/didcomm/protocol/issuecredential/states.go]: yeah, `zeroAction` is used for no action.
